ibis/pyspark/compiler.py

```diff
@@ -497,6 +507,266 @@ def compile_isinf(t, expr, scope, **kwargs):
+    def upper(v):
+        return v.str.upper()
+
+    src_column = t.translate(op.arg, scope)
+    return upper(src_column)
+
+
+@compiles(ops.Lowercase)
+def compile_lowercase(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @pandas_udf('string', PandasUDFType.SCALAR)
+    def lower(v):
+        return v.str.lower()
+
+    src_column = t.translate(op.arg, scope)
+    return lower(src_column)
+
+
+@compiles(ops.Reverse)
+def compile_reverse(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @pandas_udf('string', PandasUDFType.SCALAR)
+    def reverse(s):
+        return s.str[::-1]
+
+    src_column = t.translate(op.arg, scope)
+    return reverse(src_column)
+
+
+@compiles(ops.Strip)
+def compile_strip(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @pandas_udf('string', PandasUDFType.SCALAR)
+    def strip(s):
+        return s.str.strip()
+
+    src_column = t.translate(op.arg, scope)
+    return strip(src_column)
+
+
+@compiles(ops.LStrip)
+def compile_lstrip(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @pandas_udf('string', PandasUDFType.SCALAR)
+    def lstrip(s):
+        return s.str.lstrip()
+
+    src_column = t.translate(op.arg, scope)
+    return lstrip(src_column)
+
+
+@compiles(ops.RStrip)
+def compile_rstrip(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @pandas_udf('string', PandasUDFType.SCALAR)
+    def rstrip(s):
+        return s.str.lstrip()
+
+    src_column = t.translate(op.arg, scope)
+    return rstrip(src_column)
+
+
+@compiles(ops.Capitalize)
+def compile_capitalize(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @pandas_udf('string', PandasUDFType.SCALAR)
+    def capitalize(s):
+        return s.str.capitalize()
+
+    src_column = t.translate(op.arg, scope)
+    return capitalize(src_column)
+
+
+@compiles(ops.Substring)
+def compile_substring(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @F.udf('string')
+    def substring(s, start, length):
+        end = start + length
+        return s[start:end]
+
+    src_column = t.translate(op.arg, scope)
+    start_column = t.translate(op.start, scope)
+    length_column = t.translate(op.length, scope)
+    return substring(src_column, start_column, length_column)
+
+
+@compiles(ops.StringLength)
+def compile_string_length(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @pandas_udf('int', PandasUDFType.SCALAR)
+    def length(s):
+        return s.str.len()
+
+    src_column = t.translate(op.arg, scope)
+    return length(src_column)
+
+
+@compiles(ops.StrRight)
+def compile_str_right(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @F.udf('string')
+    def str_right(s, nchars):
+        return s[-nchars:]
+
+    src_column = t.translate(op.arg, scope)
+    nchars_column = t.translate(op.nchars, scope)
+    return str_right(src_column, nchars_column)
+
+
+@compiles(ops.Repeat)
+def compile_repeat(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    @F.udf('string')
+    def repeat(s, times):
```

Comment:
Why are we using UDFs for some of these and not others? `repeat` exists natively in spark as `F.repeat`.